### PR TITLE
AzureStorage add auth code flow

### PR DESCRIPTION
**Pull request by ostolero**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-3694)

## Description
Add support for Authorization code flow

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/AzureStorage/Integrations/AzureStorage/README.md [hunk 1]
```diff
@@ -27,19 +46,24 @@ The application must have *user_impersonation* permission and must allow public
...
+    | --- | --- | --- |
+    | Application ID |  | True |
+    | Subscription ID |  | True |
+    | Resource Group Name |  | True |
+    | Trust any certificate (not secure) |  | False |
+    | Use system proxy settings |  | False |
+    | Authentication Type | Type of authentication - could be Authorization Code flow \(recommended\) or Device Code flow. | True |
+    | Tenant ID (for user-auth mode) |  | False |
+    | Client Secret (for user-auth mode) |  | False |
+    | Client Secret (for user-auth mode) |  | False |
```
Comment (yucohen) [inline Packs/AzureStorage/Integrations/AzureStorage/README.md:62]:
```suggestion
    | Client Secret (for user-auth mode) |  | False |
```

**Review by yucohen — APPROVED** [review]
Great!

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml [hunk 2]
```diff
@@ -27,6 +27,43 @@ configuration:
...
+  required: false
+  defaultvalue:
+  type: 0
+  additionalinfo: ""
+- name: auth_code
+  display: Authorization code
+  required: false
+  defaultvalue:
+  type: 9
+  additionalinfo: For user-auth mode - received from the authorization step. See Detailed Instructions (?) section.
```
Comment (yucohen) [inline Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml:64]:
```suggestion
  additionalinfo: For user-auth mode - received from the authorization step. See Detailed Instructions section.
```

**Review by JudahSchwartz — APPROVED** [review]
Nice job finishing all of these!

**Comment by ostolero** [comment]
force merge is required as we added a new param auth_type with required: true, which fails validation but should be presented in this way.
Default is assigned in yml and in the code for bc.

**Pull request merged**